DeleteResourceJob.php

```
@@ -0,0 +1,172 @@
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Artisan;
+
+class DeleteResourceJob implements ShouldBeEncrypted, ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    public function __construct(
+        public Application|ApplicationPreview|Service|StandalonePostgresql|StandaloneRedis|StandaloneMongodb|StandaloneMysql|StandaloneMariadb|StandaloneKeydb|StandaloneDragonfly|StandaloneClickhouse $resource,
+        public bool $deleteVolumes = true,
```

PHPMD (2.15.0)

The method __construct has a boolean flag argument $deleteVolumes, which is a certain sign of a Single Responsibility Principle violation. (undefined)

(BooleanArgumentFlag)